src/Manager.cpp

```diff
@@ -70,13 +70,23 @@ namespace ClassProject {
 //         }
 //     };
 
-//     bool Manager::isVariable( BDD_ID x) {
-//         if (Table[x].label.size() == 1 && std::all_of(Table[x].label.begin(), Table[x].label.end(), ::isalpha)) {
-//             return true;
-//         }
-//         else {
-//             return false;
-//         }
-//     }
+   /**
+     * @brief isVariable
+     *
+     * checks whether the item corresponding to a given ID is a variable or not.
+     * @param x: ID of the node.
+     * @return returns true if the ID represents a variable.
+     */
+    bool Manager::isVariable( BDD_ID x) {
+        Unique_Table_Key key{};
+        auto it = Table.find(key);
```

Comment:
To find given id:
```C++
auto it = std::find_if(std::begin(Table), std::end(Table),
                         [](auto&& p) { return p->second.id == x; });
```
or
```C++
for (HandleMap::const_iterator it = Table.begin(); it != Table.end(); ++it) {
  if (it->second.it == x) ...
} 
```
See: https://stackoverflow.com/questions/32780046/finding-an-element-in-map-by-its-value

I think yours won't work for us.